sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/models/PhoneNumberErrorConverter.java

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.phonenumbers.models;
+
+import com.azure.communication.phonenumbers.implementation.models.CommunicationError;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A converter between {@link CommunicationError} and
+ * {@link PhoneNumberError}.
+ */
+public final class PhoneNumberErrorConverter {
+    /**
+     * Maps from {com.azure.communication.phonenumbrs.implementation.models.CommunicationError} to {@link PhoneNumberError}.
+     */
+    public static PhoneNumberError convert(CommunicationError obj) {
+        if (obj == null) {
+            return null;
+        }
+
+        List<PhoneNumberError> details = new ArrayList<PhoneNumberError>();
+
+        if (obj.getDetails() != null) {
+            details = obj.getDetails()
+                .stream()
+                .map(detail -> convert(detail))
+                .collect(Collectors.toList());
+        }
+
+        PhoneNumberError phoneNumberError = new PhoneNumberError(
+            obj.getMessage(),
+            obj.getCode(),
+            obj.getTarget(),
+            details
+        );
+
+        return phoneNumberError;
+    }
+
+    private PhoneNumberErrorConverter() {
```

Comment:
Can we remove this private constructor?

Reply:
I'd say we should keep it since this class is equivalent to a static class in C#. The compiler error when SDK user attempts to construct is a good thing to have.

Reply:
Maybe I am missing something. Why can't we make this a final static class and remove the private constructor? @JianpingChen